### DISCUSSION TITLE
Proxy https to localhost

### DISCUSSION THREAD

**Discussion by prcme** [mitmproxy/mitmproxy#6355] [Q&A] [answered]
I'm a beginner at mitmproxy. I want to use mitmproxy to proxy https to localhost. Is it achievable?
```py
# test.py
class PrettyHost:
    def request(self, flow):
        if flow.request.pretty_host == "https://example.org":
               flow.request.port ="3000"
               flow.request.host = "localhost"

addons = [
    PrettyHost()
]
```
```sh
mitmweb -s ./test.py
```
This way will get an error

Error message：  Server TLS handshake failed. The remote server does not speak TLS.

Thanks!

**Comment by mhils** [comment]
Yes - you either want to set `request.scheme` (https://docs.mitmproxy.org/stable/api/mitmproxy/http.html#Request.scheme) or do it all in one go and set `request.url`. :)

**Reply by prcme** [reply, marked as answer]
```py
 flow.request.url = 'http://localhost:3002' + flow.request.path 
```
Solved the proxy for my front-end static resources, 
your work is fantastic，thanks  very much
